### ISSUE TITLE
Increase the usage of expressions with combined operators

### ISSUE THREAD

**Issue by elfring** [joomla/joomla-cms#36092]
:eyes: Some source code analysis tools can help to find opportunities for improving software components.
:thought_balloon: I propose to [increase the usage of combined operators](https://www.php.net/manual/en/language.operators.assignment.php#language.operators.assignment.arithmetic "Description for assignment operators") accordingly.

Would you like to integrate anything from a transformation result which can be generated by a command like the following?
(:point_right: Please check also for questionable change suggestions because of an evolving search pattern.)
```
lokal> perl -p -i.orig -0777 -e 's#(?<target>\$\S+)\s*=\s*\k<target>[ \t]*(?<operator>[+/%^.]|-(?!>)|&(?!&)|\|(?!\|)|\*\*?|<<|>>|\?\?)#$+{target} $+{operator}=#gm' $(find ~/Aktivitäten/Joomla/lokal -name '*.php')
```

:crystal_ball: How will development interests evolve further also according to update candidates in 94 lines of this software?

**Comment by brianteeman** [comment]
This issue should be closed for obvious reasons

**Comment by elfring** [comment]
I would appreciate a more constructive feedback for proposed change possibilities.

**Comment by ReLater** [comment]
> I would appreciate a more constructive feedback for proposed change possibilities.

You are free, like everyone in the world, to provide PRs with code changes and testing instructions that others can review and test. Sometimes code reviews are sufficient, sometimes not.

If you come along with 500 changes in 48 extensions and/or libraries in 1 pr there is little chance that anyone will spend their time on it.

Likewise, the code that comes out of it should also be halfway understandable, which is not always the case with blindly used, bitwise operators, e.g. in an if(); just because it is so fancy to use one.

In short: Start and provide PRs if you think it's worth it! And others will think about if it's worth it.

**Comment by elfring** [comment]
Do you understand anything from the mentioned source code transformation approach (which could be performed also with the help the shown Perl command example)?

**Comment by ReLater** [comment]
As I've said: You cannot expect the result of a command line to be blindly merged into the Joomla core. And no one will spend their time testing all the changes made by it. So: Smaller steps are preferred. No untestable giant actions.

> Do you understand anything from the mentioned source code transformation approach

I do, my friend. But again: To let people understand it you have to provide the results as readable PHP code as PRs to a Joomla branch.

**Comment by elfring** [comment]
>You cannot expect the result of a command line to be blindly merged into the Joomla core.

I do not expect such an action.


:eyes: I am just looking for a better indication of the potential for change acceptance.

:thought_balloon: “Automatic” adjustments can occasionally be performed on demand by using advanced source code search and replacement commands, can't they?

**Comment by brianteeman** [comment]
> 👀 I am just looking for a better indication of the potential for change acceptance.

From the level of information provided - zero

**Comment by elfring** [comment]
:thought_balloon: Do any other software developers care any more for similar transformation commands (so that combined operators would be used more often)?

**Comment by rdeutz** [comment]
@elfring you got some kind advice, looking forward seeing something usefull. Till then stop wasting our time.
